### PR TITLE
NMS-14047: DCB Rest API: Parse and display cron schedule info

### PR DESCRIPTION
**Pull request by synqotik**
Parse cron schedule info from DCB Service metadata and return `nextScheduledBackup` and `scheduledInterval` in API.

Additionally:

- moved functions requiring Apache `commons-compress` from `DeviceConfigUtil` to `rest` module `CompressionUtils` class.
- Fixed device config download to have `Content-Disposition` of `attachment` so browser will download as file instead of inline text

Jira: https://issues.opennms.org/browse/NMS-14047

### CODE REVIEW

File: features/device-config/rest/src/main/java/org/opennms/features/deviceconfig/rest/impl/DefaultDeviceConfigRestService.java [hunk 1]
```diff
@@ -352,11 +356,19 @@ private void populateScheduleInfo(List<DeviceConfigDTO> dtos, Map<Integer, List<
...
 
-        return new ScheduleInfo(nextScheduledBackup, scheduleInterval);
+        try {
+            cronExpression = new CronExpression(schedulePattern);
+            cronDescription = CronExpressionDescriptor.getDescription(schedulePattern);
+        } catch (ParseException pe) {
+            return new ScheduleInfo(null, "unknown");
+        }
+
+        final Date nextScheduledBackup = cronExpression.getNextValidTimeAfter(current);
```
Comment (cgorantla) [inline features/device-config/rest/src/main/java/org/opennms/features/deviceconfig/rest/impl/DefaultDeviceConfigRestService.java:369]: Can you validate if both of these return same value ?https://github.com/OpenNMS/opennms/blob/5b67f532fc0243fd9fe9827cfe4c629eb9c5372f/features/device-config/monitor/src/main/java/org/opennms/features/deviceconfig/monitors/DeviceConfigMonitor.java#L110

Comment (synqotik) [reply features/device-config/rest/src/main/java/org/opennms/features/deviceconfig/rest/impl/DefaultDeviceConfigRestService.java:369]: I looked at source code:
https://github.com/quartz-scheduler/quartz/blob/ff348f62ece41275555b45dba5a4073c910fbeab/quartz-core/src/main/java/org/quartz/impl/triggers/CronTriggerImpl.java#L530

`trigger.getFireTimeAfter()` ultimately calls `CronExpression.getTimeAfter()` and `CronExpression.getNextTimeAfter()` just calls that method as well.

Main difference is that `Trigger` code will check its start/end times and make adjustments to make sure next fire time is within it's start/end limits, but Rest API doesn't have any knowledge of that.

I'm assuming there's a bit less overhead in calling `CronExpression.getNextValidTimeAfter` vs. building a Trigger. Kind of hard to do a unit test on this as is, but I will do some testing to confirm.

Comment (synqotik) [reply features/device-config/rest/src/main/java/org/opennms/features/deviceconfig/rest/impl/DefaultDeviceConfigRestService.java]: Updated to use the same code as in DeviceConfigMonitor, but code is copied (with a comment) in order not to add cross-dependencies.

**Review by cgorantla — APPROVED** [review]
LGTM.

**Pull request merged**